Added naked quadruple

FuriouStyles opened the pull request:
# Description
Naked Quadruple technique is exactly like the naked twins and triples except for for matching values to a cell. These cells are hard to find apparently and it's not totally clear that they area being discovered, but the code runs without errors.

Fixes # (issue)

## Type of change
Added a new technique for a total of 8

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Change Status

- [ ] Complete, tested, ready to review and merge
- [x] Complete, but not tested (may need new tests)
- [ ] Incomplete/work-in-progress, PR is for discussion/feedback

# How Has This Been Tested?

- [x] Test A
- [ ] Test B

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] My code has been reviewed by at least one peer
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] There are no merge conflicts

FuriouStyles committed:
Added naked quadruple

Algorant commented on ai.py:
```diff
@@ -248,6 +248,31 @@ def locked_triple(values):
 
+def naked_quadruple(values):
+    """ 
+    Eliminate values using the naked triple strategy.
```
line 253: should be """ ....naked quadruple strategy...""" in block comment

line 225: should be """....lock triple strategy...""" in block comment

FuriouStyles replied:
Fixed

Algorant reviewed (changes requested):
Small typo in line 225 and line 253

brucebra000 reviewed (approved):
Slight typo in the comments as stated by Ivan, but otherwise functional.

FuriouStyles committed:
Updated comments in techniques

Updated comments in techniques to show the correct technique referenced in the function
